generator.go

```diff
@@ -284,9 +284,11 @@ func newFromHash(h hash.Hash, ns UUID, name string) UUID {
 	return u
 }
 
+var netInterfaces = net.Interfaces
```

Comment:
At the call site for `netInterfaces`, we use `ifaces, err := netInterfaces()` and rely on type inference. I would prefer adding an explicit type here, in the style of these hooks: https://golang.org/pkg/internal/poll/#pkg-variables, to better communicate the signature of what is being hooked, so the reader doesn't have to go through yet another indirection.

Perhaps `var netInterfaces func() ([]net.Interface, error) = net.Interfaces`.

Reply:
I don't really like how it clutters up the code. It would be easier for my editor to give me the function header definition over trying to read that line.

I was trying to hit 100% without making the code too messy.

Reply:
I want to push back a little here. I like having the type inline, be it a little bit more cluttered as it may. Editors might help alleviate the extra indirection via highlighting, but I'd like to avoid it altogether.

Reply:
Finally getting back around to working on this PR, and this change results in a `golint` failure.

```
generator.go:287:19: should omit type func() ([]net.Interface, error) from declaration of var netInterfaces; it will be inferred from the right-hand side (golint)
var netInterfaces func() ([]net.Interface, error) = net.Interfaces
                  ^
```